src/Lock.ts

```diff
@@ -31,9 +32,11 @@ export abstract class Lock {
     acquireAttemptsLimit = defaultTimeoutOptions.acquireAttemptsLimit,
     retryInterval = defaultTimeoutOptions.retryInterval,
     refreshInterval = Math.round(lockTimeout * REFRESH_INTERVAL_COEF),
-    onLockLost = defaultOnLockLost
+    onLockLost = defaultOnLockLost,
+    externallyAcquiredIdentifier
   }: LockOptions = defaultTimeoutOptions) {
-    this._identifier = uuid4()
+    this._identifier = externallyAcquiredIdentifier || uuid4()
```

Comment:
Do I understand correctly that identifier is used for identifying "owner" of the lock?

Reply:
Yes, `refresh` updates key TTL (for mutex) only if value is "own" identifier and throws LostLockError if value is different (if value is empty (there is no owner of lock) reacquire happens).
`release` deletes key in redis only if value is "own" identifier too